projects/packages/backup/src/class-jetpack-backup-size-estimator.php

```diff
@@ -0,0 +1,316 @@
+<?php
+/**
+ * Jetpack Backup Size Estimator
+ *
+ * @package automattic/jetpack-backup-plugin
+ */
+
+// After changing this file, consider increasing the version number ("VXXX") in all the files using this namespace, in
+// order to ensure that the specific version of this file always get loaded. Otherwise, Jetpack autoloader might decide
+// to load an older/newer version of the class (if, for example, both the standalone and bundled versions of the plugin
+// are installed, or in some other cases).
+namespace Automattic\Jetpack\Backup\V0005;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit( 0 );
+}
+
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use WP_Error;
+
+/**
+ * Class Jetpack_Backup_Size_Estimator
+ *
+ * Estimates backup size by scanning local filesystem and database
+ */
+class Jetpack_Backup_Size_Estimator {
+
+	/**
+	 * Maximum execution time in seconds
+	 */
+	const DEFAULT_TIMEOUT = 25;
+
+	/**
+	 * Total file size in bytes
+	 *
+	 * @var int
+	 */
+	private $file_size = 0;
+
+	/**
+	 * Total file count
+	 *
+	 * @var int
+	 */
+	private $file_count = 0;
+
+	/**
+	 * Directory sizes breakdown
+	 *
+	 * @var array
+	 */
+	private $directory_sizes = array();
+
+	/**
+	 * Database table sizes breakdown
+	 *
+	 * @var array
+	 */
+	private $database_tables = array();
+
+	/**
+	 * Scan start time
+	 *
+	 * @var float
+	 */
+	private $start_time;
+
+	/**
+	 * Timeout in seconds
+	 *
+	 * @var int
+	 */
+	private $timeout;
+
+	/**
+	 * Constructor
+	 *
+	 * @param int $timeout Maximum execution time in seconds.
+	 */
+	public function __construct( $timeout = self::DEFAULT_TIMEOUT ) {
+		$this->start_time = microtime( true );
+		$this->timeout    = $timeout;
+	}
+
+	/**
+	 * Estimate backup size
+	 *
+	 * @return array|WP_Error
+	 */
+	public function estimate_size() {
+		$this->start_time = microtime( true );
+
+		$directories = $this->get_scan_directories();
+
+		foreach ( $directories as $label => $path ) {
+			if ( $this->is_timeout_approaching() ) {
+				break;
+			}
+
+			$this->scan_directory( $path, $label );
+		}
+
+		if ( ! $this->is_timeout_approaching() ) {
+			$this->get_database_size();
+		}
+
+		$database_size    = $this->get_total_database_size();
+		$processing_time  = microtime( true ) - $this->start_time;
+		$timeout_occurred = $this->is_timeout_approaching();
+
+		return array(
+			'file_size'       => $this->file_size,
+			'database_size'   => $database_size,
+			'total_size'      => $this->file_size + $database_size,
+			'directories'     => $this->directory_sizes,
+			'database_tables' => $this->database_tables,
+			'file_count'      => $this->file_count,
+			'database_count'  => $this->get_database_count(),
```

Comment:
Isn't this field redundant (it just counts entries in the `database_tables` field)?